src/nickel.rs

```diff
@@ -347,25 +346,28 @@ mod tests {
     use std::str::FromStr;
     use std::net::SocketAddr;
 
-    #[test]
+    #[tokio::test]
     #[should_panic(expected = "invalid socket address")]
-    fn invalid_listen_addr() {
-        Nickel::new().listen("127.0.0.1.6667").unwrap();
+    async fn invalid_listen_addr() {
+        Nickel::new().listen("127.0.0.1.6667").await.unwrap();
     }
 
-    #[test]
-    fn can_get_server_address() {
-        let server = Nickel::new().listen("127.0.0.1:12345").unwrap();
+    // TODO: These tests are commented out since we no longher provide access to
+    // the underlying server. Do we need to?
 
-        assert_eq!(server.socket(), SocketAddr::from_str("127.0.0.1:12345").unwrap());
-        server.detach();
-    }
+    // #[test]
+    // fn can_get_server_address() {
+    //     let server = Nickel::new().listen("127.0.0.1:12345").await.unwrap();
 
-    #[test]
-    fn can_get_server_address_with_random_port() {
-        let server = Nickel::new().listen("127.0.0.1:0").unwrap();
+    //     assert_eq!(server.socket(), SocketAddr::from_str("127.0.0.1:12345").unwrap());
```

Comment:
This isn't a random port.

Reply:
Commented out with notes to determine what testing/features are needed as part of further migration cleanup.